trpcultivate_phenotypes/src/Plugin/Validators/ValidDelimitedFile.php

```diff
@@ -0,0 +1,212 @@
+   */
+  public function validateRawRow($raw_row) {
+    
+    // Parameter check, verify that raw row is not an empty string value.
+    if (empty(trim($raw_row))) {
+      return [
+        'case' => 'Raw row is empty',
+        'valid' => FALSE,
+        'failedItems' => ['raw_row' => 'is an empty string value']
+      ];
+    }
+
+    // Check if the line has some delimiters used (only if number of expected columns is greater than 1).
+    $expected_columns = $this->getExpectedColumns();
+    if ($expected_columns['number_of_columns'] == 1) {
+      return [
+        'case' => 'Data file raw row is delimited',
+        'valid' => TRUE,
+        'failedItems' => []
+      ];
+    }
+
+    // Specifically, check the line includes at least one of the delimiters returned by the get file delimiter method. 
+    $input_file_mime_type = $this->getFileMimeType();
+    $input_file_type_delimiters = $this->getFileDelimiters($input_file_mime_type);
+
+    $delimiters_used = [];
+    foreach($input_file_type_delimiters as $delimiter) {
+      if (strpos($raw_row, $delimiter)) {
+        array_push($delimiters_used, $delimiter);
+      }
+    }
+      
+    // Not one of the supported delimiters was detected in the raw row.
+    if (empty($delimiters_used)) {
+      return [
+        'case' => 'None of the delimiters supported by the file type was used',
+        'valid' => FALSE,
+        'failedItems' => ['raw_row' => $raw_row]
+      ];
+    }
+
+    // Split the line and see if the number of values returned equals to the expected number of columns.
+    // Use the strict flag of the validator columns configuration to compare the columns returned by
+    // split method and the configured columns.
+
+    // A strict flag set to True means exact match whereas False requires at least the configured columns.
+    $delimiters_checked = [];
+    foreach($delimiters_used as $delimiter) {
+      $columns = TripalCultivatePhenotypesValidatorBase::splitRowIntoColumns($raw_row, $input_file_mime_type);
+
+      if ($expected_columns['strict']) {
+        // A strict comparison - exact match only.
+        if (count($columns) != $expected_columns['number_of_columns']) {
+          array_push($delimiters_checked, $delimiter);
+        }
+      }
+      else {
+        // Not a strict comparison - at least x number of columns.
+        if (count($columns) < $expected_columns['number_of_columns']) {
+          array_push($delimiters_checked, $delimiter);
+        }
+      }
+    }
+
+    // If all delimiters failed the checks, then the line failed due to none of the
+    // delimiter was able to split the line into the expected number of columns.
+    if ($delimiters_used == $delimiters_checked) {
+      return [  
+        'case' => 'Raw row is not delimited',
+        'valid' => FALSE,
+        'failedItems' => ['raw_row' => $raw_row]
+      ];
+    }
+
+    return [
+      'case' => 'Data file raw row is delimited',
```

Comment:
To make this consistent with line 145, this should be
```suggestion
      'case' => 'Raw row is delimited',
```